FWCore/Catalog/src/FileLocator.cc

```diff
@@ -41,23 +40,34 @@ namespace {
 
 }  // namespace
 
+namespace pt = boost::property_tree;
+
 namespace edm {
+
   FileLocator::FileLocator(std::string const& catUrl, unsigned iCatalog) : m_destination("any") {
-    init(catUrl, iCatalog);
+    init_trivialCatalog(catUrl, iCatalog);
+  }
 
-    // std::cout << m_protocols.size() << " protocols" << std::endl;
-    // std::cout << m_directRules[m_protocols[0]].size() << " rules" << std::endl;
+  FileLocator::FileLocator(edm::CatalogAttributes const& catAttr, unsigned iCatalog) : m_destination("any") {
+    init(catAttr, iCatalog);
   }
 
   FileLocator::~FileLocator() {}
 
-  std::string FileLocator::pfn(std::string const& ilfn) const { return convert(ilfn, m_directRules, true); }
-
-  std::string FileLocator::lfn(std::string const& ipfn) const { return convert(ipfn, m_inverseRules, false); }
+  std::string FileLocator::pfn(std::string const& ilfn, edm::CatalogType catType) const {
+    if (catType == edm::CatalogType::TrivialCatalog)
+      return convert(ilfn, m_directRules_trivialCatalog, true);
+    return convert(ilfn, m_directRules, true);
+  }
 
   std::string FileLocator::convert(std::string const& input, ProtocolRules const& rules, bool direct) const {
     std::string out = "";
-
+    if (!m_prefix.empty()) {
+      out =
+          m_prefix + "/" +
+          input;  //need "/" since not all inputs have it (normally logical file name is "/store/..."), for example teststreamfile.dat in "int readInvalidLFN(bool verbose)" of ReadStreamerFile.cpp
```

Comment:
This could check if `input[0]` == '/' and not add the `/` if that is the case. In any case could you move the comment above the line so that the code looks better after `clang-format` ?

Reply:
Done